### PR TITLE
Add documentation for `cupyx.optimizing.optimize`

### LINKED ISSUE #3383: Write docs for Optuna optimization

### PR DESCRIPTION
**Pull request by asi1024**
Close #3383.

### CODE REVIEW

**Comment by kmaehashi** [comment]
Could you also add a new docs page, like `docs/source/ereference/optimize.rst`?

**Comment by asi1024** [comment]
PTAL.

File: docs/source/reference/optimize.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Automatic Kernel Parameters Optimizations
+=========================================
+
```
Comment (kmaehashi) [inline docs/source/reference/optimize.rst:3]: Could you add some more information here?

* Optuna installation is required
* Currently it works for reduction ops only

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
LGTM!

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by leofang** [comment]
Not sure if related: https://github.com/sphinx-doc/sphinx/issues/7807. Making `sphinx<=3.0.4` would likely fix it.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: docs/source/reference/optimize.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Automatic Kernel Parameters Optimizations
```
Comment (leofang) [inline docs/source/reference/optimize.rst:1]: How about this?
```suggestion
.. module:: cupyx.optimizing

Automatic Kernel Parameters Optimizations
```

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by leofang** [comment]
Still not working...I guess you need to add `optimizing` to `docs/source/reference/ext.rst`?

**Comment by kmaehashi** [comment]
`cupyx.optimize` was not listed in `setup.py`. @asi1024 will fix it.

**Comment by kmaehashi** [comment]
This is also needed https://github.com/cupy/cupy/pull/3427

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Merged https://github.com/chainer/chainer-test/pull/578 for the doctest.
Jenkins, test this please.

**Pull request merged**